py-polars/tests/unit/io/database/test_read.py

```diff
@@ -479,6 +479,7 @@ def test_read_database_mocked(
         assert isinstance(res, GeneratorType)
         res = pl.concat(res)
 
+    res = cast(pl.DataFrame, res)
```

Comment:
Typeguard could be a bit of a more programmatic way of doing this, but it's just a test, so I figure this suffices.